scripts/pack.mjs

```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env node
+// -*- js -*-
+
+"use strict";
+
+
+import { promises as fs } from "fs";
+import extract from 'extract-zip';
+
+import fsync from "fs";
+// import info from "../package.json";
+import path from "path";
+import { minify } from "uglify-js";
+import https from 'https'
+import zlib from 'zlib';
+
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+if (!fsync.existsSync('./dist'))
+    await fs.mkdir('./dist')
+if (!fsync.existsSync('./tmp'))
+    await fs.mkdir('./tmp')
+
+const libFiles = await readFiles('./lib');
+const srcFiles = await readFiles('./src');
+
+// console.log(libFiles)
+
+if (!fsync.existsSync('./tmp/assets.zip')) {
+    console.log('Downloading assets')
+    const download = new Promise(resolve => {
+
+        https.get('https://www.astralvault.net/games/SA/MapEditor/assets.zip', async res => {
+
+            const path = './tmp/assets.zip';
+            const filePath = fsync.createWriteStream(path);
+            res.pipe(filePath);
+            filePath.on('finish', () => {
+                filePath.close();
+                console.log('Download Completed');
+                resolve();
+            })
+        })
+    });
+    await download;
+}
+
+
+console.log('extracting assets...')
+await extract('./tmp/assets.zip', { dir: path.join(__dirname, '..', 'dist', 'assets') })
+console.log('assets extracted')
+
+console.log('minifiying...')
+const minified = minify(libFiles.concat(srcFiles))
+
+if (minified.warnings)
+    console.warn(minified.warnings)
+if (minified.error)
+    console.warn(minified.error)
+
+await fs.writeFile('./dist/TI3SA.min.js', minified.code);
+if (minified.map)
+    await fs.writeFile('./dist/TI3SA.min.js.map', minified.map);
```

Comment:
What is this for?

Reply:
When adding source map it writes it to the disk.

So if someone would enable source mapping it would actually land on the disk.

I think I can just enable it...